src/app/shared/editor/components/yaml-editor/yaml-editor.component.ts

```diff
@@ -0,0 +1,43 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    EventEmitter,
+    Input,
+    OnChanges,
+    Output,
+    SimpleChanges,
+} from "@angular/core";
+
+import * as monaco from "monaco-editor";
+import { MaybeNull } from "../../../../common/app.types";
+import { MonacoService } from "../../services/monaco.service";
+import { YAML_EDITOR_OPTIONS } from "./config-editor.events";
+
+@Component({
+    selector: "app-yaml-editor",
+    templateUrl: "./yaml-editor.component.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class YamlEditorComponent implements OnChanges {
+    @Input() public yamlTemplate = "";
+    @Input() public yamlError: MaybeNull<string>;
+    @Input() public readonly YAML_EDITOR_OPTIONS: monaco.editor.IStandaloneEditorConstructionOptions =
+        YAML_EDITOR_OPTIONS;
+
+    @Output() public yamlTemplateChange = new EventEmitter<string>();
+    @Output() public onRunSql = new EventEmitter<null>();
```

Comment:
I guess YAML editor will not allow to "Run SQL" :-)

Reply:
removed